packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -53,6 +53,9 @@ enum SelectionResult {
   /// [SelectAllSelectionEvent], [ClearSelectionEvent], and
   /// [SelectWordSelectionEvent].
   none,
+  /// The selection may begin in this [Selectable], further selection should
+  /// continue to next [Selectable] in screen order.
+  forward,
```

Comment:
Still working on these docs.

Reply:
What's the difference between next and forward?

Reply:
I wanted to handle these two scenarios:
1. A paragraph is broken by a `WidgetSpan`, we are in the `Selectable` containing the first half of the paragraph before the `WidgetSpan` so walk to the next selectable to continue searching for the complete paragraph. In this case the delegate should set the `currentSelectionStartIndex` and continue walking through the selectable tree.
2. The paragraph range comes after the range contained by the `Selectable` so walk to the next selectable. In this case we continue walking through the selectable tree but the delegate should not set `currentSelectionStartIndex`.

I wanted some way to tell the delegate that the selection begins in the receiver of the event and we should continue walking through the selectable tree to find the rest of the selection. 

From the documentation of `SelectionResult.next`
`There is nothing left to select forward in this Selectable, and further selection should extend to the next Selectable in screen order.`
it sounds like it would be appropriate in this case, but then that would make `currentSelectionStartIndex` inaccurate since there is no way to differentiate between the two scenarios.